src/Microsoft.Fx.Portability/InternalServerErrorException.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Fx.Portability.Resources;
+using System;
+
+namespace Microsoft.Fx.Portability
+{
+    public class InternalServerErrorException : PortabilityAnalyzerException
+    {
+        private InternalServerErrorException(string message, string exception)
+            : base(string.Format(LocalizedStrings.InternalServerErrorMessage, message, exception))
+        { }
+
+        /// <summary>
+        /// This returns a properly formatted exception received from the server.
+        /// </summary>
+        public static InternalServerErrorException Create(string message, string exception)
+        {
+            var split = exception.Split(new[] { "\\r\\n", "\\r", "\\n" }, StringSplitOptions.RemoveEmptyEntries);
+
+            return new InternalServerErrorException(message, string.Join(Environment.NewLine, split));
```

Comment:
Why are you removing newlines and then adding them back? is it to get a consistent newline character? Many times, this will be occurring on the service side, so Environment.NewLine will be from the server and may differ from what a client expects.